pyomo/contrib/solver/solvers/ipopt.py

```diff
@@ -499,46 +539,140 @@ def solve(self, model, **kwds):
+                    iter_data['restoration'] = iter_data['iter'].endswith('r')
+                    if iter_data['restoration']:
+                        iter_data['iter'] = iter_data['iter'][:-1]
+                    assert str(iterations) == iter_data.pop(
+                        'iter'
+                    ), f"Number of iterations ({iterations}) does not match the "
+                    "parsed row in the iterations table"
+
+                    # Separate alpha_pr into numeric part and optional tag
+                    iter_data['step_acceptance'] = iter_data['alpha_pr'][-1]
+                    if iter_data['step_acceptance'] in _ALPHA_PR_CHARS:
+                        iter_data['alpha_pr'] = iter_data['alpha_pr'][:-1]
+                    else:
+                        iter_data['step_acceptance'] = None
+
+                    # Attempt to cast all values to float where possible
+                    for key in columns:
+                        if key == 'iter':
+                            continue
+                        if iter_data[key] == '-':
+                            iter_data[key] = None
+                        else:
+                            try:
+                                iter_data[key] = float(iter_data[key])
+                            except (ValueError, TypeError):
+                                logger.warning(
+                                    "Error converting Ipopt log entry to "
+                                    f"float:\n\t{sys.exc_info()[1]}\n\t{line}"
+                                )
+
+                    all_iterations.append(iter_data)
+                    iterations += 1
+
+            parsed_data['iteration_log'] = all_iterations
+
+        # Extract scaled and unscaled table
+        scaled_unscaled_match = re.findall(
+            r'Objective\.*:\s*([-+eE0-9.]+)\s+([-+eE0-9.]+)\s*'
+            r'Dual infeasibility\.*:\s*([-+eE0-9.]+)\s+([-+eE0-9.]+)\s*'
+            r'Constraint violation\.*:\s*([-+eE0-9.]+)\s+([-+eE0-9.]+)\s*'
+            # Next field is optional because it shows up in new-style ipopt output, but not old style
+            r'(?:Variable bound violation: *([-+eE0-9.]+) *([-+eE0-9.]+) *\s*)?'
+            r'Complementarity\.*:\s*([-+eE0-9.]+)\s+([-+eE0-9.]+)\s*'
+            r'Overall NLP error\.*:\s*([-+eE0-9.]+)\s+([-+eE0-9.]+)',
+            output,
+            re.DOTALL,
+        )
+        if scaled_unscaled_match:
+            fields = [
+                "incumbent_objective",
+                "dual_infeasibility",
+                "constraint_violation",
+                "complementarity_error",
+                "overall_nlp_error",
+            ]
+            scaled = {
+                k: float(v)
+                for k, v in zip(fields, scaled_unscaled_match[0][0:10:2])
+                if v
+            }
+            unscaled = {
+                k: float(v)
+                for k, v in zip(fields, scaled_unscaled_match[0][1:10:2])
```

Comment:
The slice upper bound is now incorrect: it needs to be `[1:12:2]` (or just `[1::2]`)